lib/features/support/servlet.rb

```diff
@@ -40,6 +39,19 @@ def do_POST(request, response)
     end
     response.header['Access-Control-Allow-Origin'] = '*'
     response.status = Server.status_code
+  rescue JSON::ParserError => e
+    msg = "Unable to parse request as JSON: #{e.message}"
+    $logger.error msg
+    Server.invalid_requests << {
```

Comment:
Does this have implications for failed requests from mapping file uploads, etc? Those requests aren't JSON.

Reply:
It shouldn't do.  The `JSON::ParserError` rescue block is redundant for mapping file uploads, as that arm of the `case` won't try to parse it as JSON.  The `StandardError` block seemed like a good thing to put in place for everything, so I thought the code was neatest by having both handlers together, even if one isn't strictly necessary for some of the logic.

Reply:
Ah! I see it now. lgtm.